### PR TITLE
fix: Functional Tests with Encryption/HTTPS, S3

### PR DESCRIPTION
**Pull request by BigUstad**
And miscellaneous code refactoring, obsolete declarations.

### CODE REVIEW

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 1]
```diff
@@ -1231,30 +1233,41 @@ internal async static Task CopyObject_Test2(MinioClient minio)
...
                 }
                 catch (MinioException ex)
                 {
-                    Console.WriteLine(ex.Message);
-                    StringAssert.Equals(ex.Message, "MinIO API responded with message=At least one of the pre-conditions you specified did not hold");
+                    Assert.IsTrue(ex.Message.Contains("MinIO API responded with message=At least one of the pre-conditions you specified did not hold"));
                 }
 
-                RemoveObjectArgs rmArgs = new RemoveObjectArgs()
+                RemoveObjectArgs rmArgs1 = new RemoveObjectArgs()
```
Comment (kannappanr) [inline Minio.Functional.Tests/FunctionalTest.cs:1242]: do we not have `RemoveObjects`, why are you making multiple calls?

**Review by kannappanr — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 1 above)
Comment (BigUstad) [reply Minio.Functional.Tests/FunctionalTest.cs:1242]: `RemoveObjectsArgs` is implemented for different objects in 1 bucket. In this case it is 2 objects in different buckets.

File: Minio/DataModel/ObjectOperationsArgs.cs [hunk 2]
```diff
@@ -1160,82 +1338,249 @@ public override RestRequest BuildRequest(RestRequest request)
...
     {
-        public NewMultipartUploadArgs()
+        internal bool ReplaceMetadataDirective { get; set; }
+        internal bool ReplaceTagsDirective { get; set; }
+        internal string StorageClass { get; set; }
+        internal ObjectStat SourceObjectInfo { get; set; }
+        internal CopySourceObjectArgs SourceObject { get; set; }
+        internal RetentionMode ObjectLockRetentionMode { get; set; }
+        internal DateTime RetentionUntilDate { get; set; }
+        internal bool ObjectLockSet { get; set; }
```
Comment (poornas) [inline Minio/DataModel/ObjectOperationsArgs.cs:1369]: why would NewMultipartUpload args reference CopySourceObjectArgs, ReplaceMetadataDirective etc - this is part of CopyObjectArgs/MultipartCopyUploadArgs isn't it?

**Review by poornas — CHANGES_REQUESTED** [review]

File: Minio/DataModel/ILM/AbortIncompleteMultipartUpload.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@
 using System;
 using System.Xml.Serialization;
 
-namespace Minio.DataModel
+namespace Minio.DataModel.ILM
```
[the remark this answers is not in the document]
Comment (BigUstad) [reply Minio/DataModel/ILM/AbortIncompleteMultipartUpload.cs:20]: Yes. This class is now only used in `LifecyclerRule`. 
As per an earlier discussion, if something is used in Lifecycle it goes in ILM namespace. If it gets used elsewhere, there will be another one in a different namespace.

**Review by poornas — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]

**Pull request merged**